lib/ansible/modules/cloud/pubnub/pubnub_blocks.py

```diff
@@ -0,0 +1,2090 @@
+
+# Change event handler trigger event type.
+- name: Change event handler 'event'
+  pubnub_blocks:
+    email: '{{ email }}'
+    pwd: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    block: '{{ block_name }}'
+    event_handlers:
+      -
+        name: '{{ handler_name }}'
+        event: 'js-after-publish'
+
+# Stop block and event handlers.
+- name: Stopping block
+  pubnub_blocks:
+    email: '{{ email }}'
+    pwd: '{{ password }}'
+    application: '{{ app_name }}'
+    keyset: '{{ keyset_name }}'
+    block: '{{ block_name }}'
+    state: stop
+
+# Multiple module calls with cached result passing
+- name: Create '{{ block_name }}' block
+      register: module_cache
+      pubnub_blocks:
+        email: '{{ email }}'
+        pwd: '{{ password }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        block: '{{ block_name }}'
+        state: present
+    - name: Add '{{ event_handler_1_name }}' event handler to '{{ block_name }}'
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        block: '{{ block_name }}'
+        state: present
+        event_handlers:
+          -
+            src: '{{ path_to_handler_1_source }}'
+            name: '{{ event_handler_1_name }}'
+            channels: '{{ event_handler_1_channel }}'
+            event: 'js-before-publish'
+    - name: Add '{{ event_handler_2_name }}' event handler to '{{ block_name }}'
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        block: '{{ block_name }}'
+        state: present
+        event_handlers:
+          -
+            src: '{{ path_to_handler_2_source }}'
+            name: '{{ event_handler_2_name }}'
+            channels: '{{ event_handler_2_channel }}'
+            event: 'js-before-publish'
+    - name: Start '{{ block_name }}' block
+      register: module_cache
+      pubnub_blocks:
+        cache: '{{ module_cache }}'
+        application: '{{ app_name }}'
+        keyset: '{{ keyset_name }}'
+        block: '{{ block_name }}'
+        state: start
+'''
+
+RETURN = '''
+module_cache:
+  description: Cached account information. In case if with single play module used few times it is better to pass \
+               cached data to next module calls to speed up process.
+  type: dict
+'''
+
+PUBNUB_API_URL = "https://admin.pubnub.com/api"
+PN_BLOCK_STATE_CHECK_INTERVAL = 1
+PN_BLOCK_STATE_CHECK_MAX_COUNT = 30
+# Import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import *
+from ansible.module_utils.urls import *
+PN_URLENCODE = None
+try:
+    import urllib.parse
+    PN_URLENCODE = urllib.parse.urlencode
+except ImportError:
+    import urllib
+    PN_URLENCODE = urllib.urlencode
```

Comment:
To import urlencode, you can use this line:

```
from ansible.compat.module_utils.six.moves.urllib.parse import urlencode
```
Then for the rest of the file just use `urlencode` as normal.